tools/create_service_bundle.py

```diff
@@ -0,0 +1,568 @@
+#!/usr/bin/env python3
+#
+# TODO: usage, description.
+
+from datetime import date, datetime
+from typing import List
+import json
+import logging
+import os
+import re
+import retrying
+import sys
+
+from sdk_utils import groupby
+import sdk_cmd
+import sdk_diag
+import sdk_utils
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+
+
+DEFAULT_RETRY_WAIT = 1000
+DEFAULT_RETRY_MAX_ATTEMPTS = 5
+
+
+@retrying.retry(
+    wait_fixed=DEFAULT_RETRY_WAIT, stop_max_attempt_number=DEFAULT_RETRY_MAX_ATTEMPTS
+)
+def _get_task_page(
+    offset: int, limit: int, framework_id: str = None, task_id: str = None
+) -> dict:
+    path = "/mesos/tasks?offset={}&limit={}".format(offset, limit)
```

Comment:
Would it not make sense to rather download `state.json` once at some point and operate on that. Otherwise we may have changing state while performing these requests.

Reply:
Yes! The script gets tasks through the cluster state with the latest changes.